SingularityUI/app/models/Request.coffee

```diff
@@ -235,7 +238,46 @@ class Request extends Model
                 if !duration or (duration and @_validateDuration(duration, @promptPause, callback))
                     @pause(killTasks, duration, message).done callback
 
-    promptScale: (callback) =>
+    callScale: (data, bounce, incremental, message, duration, callback) =>
+        @scale(data).done =>
+            if bounce
+                @bounce({incremental}).done callback
+            else
+                callback()
+
+    promptScaleEvenNumberRacks: (scaleData) =>
+        vex.dialog.open
+            message: scaleEvenNumbersTemplate
+                instances: parseInt(scaleData.data.instances)
+                notOneInstance: parseInt(scaleData.data.instances) != 1
+                racks: @racks.length
+                notOneRack: @racks.length != 1
+                mod: scaleData.mod
+                modNotOne: scaleData.mod != 1
+                lower: parseInt(scaleData.data.instances) - scaleData.mod
+                higher: parseInt(scaleData.data.instances) + @racks.length - scaleData.mod
+                config: config
+            input: """
+                
+            """
+            buttons: [
+                $.extend _.clone(vex.dialog.buttons.YES), text: "Scale"
+                vex.dialog.buttons.NO
+            ]
+            scaleData: scaleData # Not sure why this is necessary, callback for whatever reason doesn't have access to the function's variables
+            callback: (data) =>
+                return unless data
+                scaleData.data.instances = data.instances
+                @callScale scaleData.data, scaleData.bounce, scaleData.incremental, scaleData.message, scaleData.duration, scaleData.callback
+
+    checkScaleEvenNumberRacks: (data, bounce, incremental, message, duration, callback) =>
+        return 0 unless @attributes.request.rackSensitive
+        if not @racks
+            @racks = new Racks []
+            @racks.fetch { async: false }
```

Comment:
We should be sure to catch errors here. If the call to fetch the number of racks fails for some reason, we should allow the user to continue without the extra prompt